source/howto/pipelines/tasks.rst

```diff
@@ -70,3 +70,76 @@ All the outputs of the Task will be passed to the next node.
             - [train.output.model*, evaluate.input.model]
     ..
 
+Dynamically defining a Task inside a pipeline
+---------------------------------------------
+
+You can use an execution node to parametrize a downstream task node's variations by printing out metadata and defining an edge that connects the metadata to the Task's parameter.
+
+List-shaped metadata from executions automatically "spread" into a Multiple variant parameter if the destination node is a task.
+
+See the example below where we have two files:
+
+1) ``preprocess.py`` will generate a list of store IDs
+2) ``train.py`` will train a model for each store, using the store ID that it gets
+
+``preprocess.py``
+
+.. code-block:: python
+
+    import json
+
+    # Your logic to fetch the right values...
+
+    storeids = [122,154,209,916,345]
+
+    # Print out a list as metadata
+    print(json.dumps({
+        "storeids": storeids
+    }))
+
+``train.py``
+
+.. code-block:: python
+
+    import valohai
+
+    storeid = valohai.parameters('id').value
+
+    # train model...
+
+And the ``valohai.yaml`` that defines our steps, and the pipeline.
+In the pipeline's edge we pass the ``storeids`` from ``preprocess`` to the ``train`` Task's ``id`` parameter. This will create a ``train`` task with a set of multiple values for the ``id`` parameter (in our example, a Task with 5 executions, each with it's own ID parameter value)
```

Comment:
```suggestion
In the pipeline's edge we pass the ``storeids`` from ``preprocess`` to the ``train`` Task's ``id`` parameter. This will create a ``train`` Task with a set of multiple values for the ``id`` parameter (in our example, a Task with 5 executions, each with it's own ID parameter value)
```